lib/improver/nowcasting/optical_flow.py

```diff
@@ -0,0 +1,873 @@
+            return float(np.diff((new_coord).points)[0])
+
+        grid_vel_x = self.vel_x.data / grid_spacing(cube.coord(axis="x"))
+        grid_vel_y = self.vel_y.data / grid_spacing(cube.coord(axis="y"))
+
+        # perform advection and create output cube
+        advected_data = self._advect_field(cube.data, grid_vel_x, grid_vel_y,
+                                           timestep.total_seconds(),
+                                           fill_value)
+        advected_cube = cube.copy(data=advected_data)
+
+        # increment output cube time
+        original_datetime, = \
+            (cube.coord("time").units).num2date(cube.coord("time").points)
+        new_datetime = original_datetime + timestep
+        new_time = (cube.coord("time").units).date2num(new_datetime)
+        advected_cube.coord("time").points = new_time
+
+        return advected_cube
+
+
+class OpticalFlow(object):
+    """
+    Class to calculate advection velocities along two orthogonal spatial axes
+    from time-separated fields using an optical flow algorithm
+    """
+
+    def __init__(self, data_smoothing_radius=7., data_smoothing_method='box',
+                 boxsize=30., point_weight=0.1, iterations=100):
+        """
+        Initialise the class with smoothing parameters for estimating gridded
+        u- and v- velocities via optical flow.
+
+        input:
+            data_smoothing_radius (float):
+                Kernel radius in km over which to smooth input data before
+                estimating partial derivatives.  Should not be less than 3x
+                the grid length of the input data cube.
+            data_smoothing_method (str):
+                Smoothing method to be used on input fields before estimating
+                partial derivatives.  Can be square 'box' (as used in STEPS) or
+                circular 'kernel' (used in post-calculation smoothing).
+            boxsize (float):
+                Square box size in km over which all data points are assumed
+                to have the same velocity to enable matrix inversion
+                (solve_for_uv()).  Should not be less than 3x the grid length
+                of the input data cube.  Must be larger than the kernel radius
+                used to smooth the input data fields.
+            point_weight (float):
+                Weight given to the velocity of a point (box), as opposed to
+                its neighbours, when doing the smart smoothing after velocity
+                calculation.  Values 0-1.
+            iterations (int):
+                Number of iterations to perform in post-calculation smoothing.
+
+        Martina's TODO: need to calculate a suitable kernel size given dt (what
+        is expected movement between slices) and expected velocities
+        """
+
+        # parameters for input data smoothing
+        self.data_smoothing_radius_km = data_smoothing_radius
+        self.data_smoothing_radius = None
```

Comment:
As far as I understand it - we set this value to "None" now and later on in line 829 we recalculate this value in grid square units using "data_smoothing_radius_km".
I think this could be clearer, we can either:
1) Add a comment to this line to make it clearer.
2) Or perhap change the variables around.  Within __init__ we define data_smoothing_radius_km=7 instead of data_smoothing_radius=7.